cmd/flags/flags.go

```diff
@@ -53,21 +55,29 @@ func TryLoadConfigFile(v *viper.Viper) error {
 type SharedFlags struct {
 	// Logging holds logging configuration
 	Logging logging
+	// HealthCheck holds health check configuration
+	HealthCheck healthCheck
 }
 
 type logging struct {
 	Level string
 }
 
+type healthCheck struct {
+	Port int
+}
+
 // AddFlags adds flags for SharedFlags
 func AddFlags(flagSet *flag.FlagSet) {
 	flagSet.String(spanStorageType, "", fmt.Sprintf("Deprecated; please use %s environment variable", storage.SpanStorageTypeEnvVar))
 	flagSet.String(logLevel, "info", "Minimal allowed log Level. For more levels see https://github.com/uber-go/zap")
+	flagSet.Int(healthCheckHTTPPort, 0, "The http port for the health check service")
```

Comment:
The default port is removed since it's difficult to pass in the current interface. So I let NewHealthCheck set it when each component initializes health check.
Do we need to add a default port in help message again? Maybe like below..
```go
flagSet.Int(healthCheckHTTPPort, 0, "The http port for the health check service. For default port, see https://https://www.jaegertracing.io/XXXXX")
```